spec/unit/classes/client_spec.rb

```diff
@@ -12,10 +12,20 @@
   describe 'with parameters' do
     let :params do
       {
-        :validcon_script_path  => '/opt/bin/my-validate-con.sh',
-        :package_ensure => 'absent',
-        :package_name   => 'mypackage',
-        :file_ensure    => 'file'
+        :validcon_script_path => '/opt/bin/my-validate-con.sh',
+        :package_ensure       => 'absent',
+        :package_name         => 'mypackage',
+        :file_ensure          => 'file',
+        :validate_connections => { :test => {
+                                              :database_host => 'test',
```

Comment:
Try changing symbols to strings throughout this validate_connections hash, like you've got in server_spec.rb. I don't think rspec-puppet can convert hash keys properly in a manifest.